File: mRemoteV1/CS/Tools/ReconnectGroup.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports


namespace mRemoteNG
{
	public partial class ReconnectGroup
	{
		public ReconnectGroup()
		{
			InitializeComponent();
		}
		private bool _ServerReady;
public bool ServerReady
		{
			get
			{
				return _ServerReady;
			}
			set
			{
				if (value == true)
				{
					SetStatusImage(global::My.Resources.HostStatus_On);
				}
				else
				{
					SetStatusImage(global::My.Resources.HostStatus_Off);
				}
				
				_ServerReady = value;
			}
		}
		
		private delegate void SetStatusImageCB(Image Img);
		private void SetStatusImage(Image Img)
		{
			if (pbServerStatus.InvokeRequired)
			{
				SetStatusImageCB d = new SetStatusImageCB(SetStatusImage);
				ParentForm.Invoke(d, new object[] {Img});
			}
			else
			{
				pbServerStatus.Image = Img;
			}
		}
		
		public void chkReconnectWhenReady_CheckedChanged(System.Object sender, System.EventArgs e)
		{
			_ReconnectWhenReady = chkReconnectWhenReady.Checked;
		}
		
		private bool _ReconnectWhenReady;
public bool ReconnectWhenReady
		{
			get
			{
				return _ReconnectWhenReady;
			}
			set
			{
				_ReconnectWhenReady = value;
				SetCheckbox(value);
			}
		}
		
		private delegate void SetCheckboxCB(bool Val);
		private void SetCheckbox(bool Val)
		{
			if (chkReconnectWhenReady.InvokeRequired)
			{
				SetCheckboxCB d = new SetCheckboxCB(SetCheckbox);
				ParentForm.Invoke(d, new object[] {Val});
			}
			else
			{
				chkReconnectWhenReady.Checked = Val;
			}
		}
		
		public delegate void CloseClickedEventHandler();
		private CloseClickedEventHandler CloseClickedEvent;
		
		public event CloseClickedEventHandler CloseClicked
		{
			add
			{
				CloseClickedEvent = (CloseClickedEventHandler) System.Delegate.Combine(CloseClickedEvent, value);
			}
			remove
			{
				CloseClickedEvent = (CloseClickedEventHandler) System.Delegate.Remove(CloseClickedEvent, value);
			}
		}
		
		
		public void btnClose_Click(System.Object sender, System.EventArgs e)
		{
			if (CloseClickedEvent != null)
				CloseClickedEvent();
		}
		
		public void tmrAnimation_Tick(System.Object sender, System.EventArgs e)
		{
			switch (lblAnimation.Text)
			{
				case "":
					lblAnimation.Text = "»";
					break;
				case "»":
					lblAnimation.Text = "»»";
					break;
				case "»»":
					lblAnimation.Text = "»»»";
					break;
				case "»»»":
					lblAnimation.Text = "";
					break;
			}
		}
		
		private delegate void DisposeReconnectGroupCB();
		public void DisposeReconnectGroup()
		{
			if (this.InvokeRequired)
			{
				DisposeReconnectGroupCB d = new DisposeReconnectGroupCB(DisposeReconnectGroup);
				ParentForm.Invoke(d);
			}
			else
			{
				this.Dispose();
			}
		}
		
		public void ReconnectGroup_Load(object sender, System.EventArgs e)
		{
			ApplyLanguage();
		}
		
		private void ApplyLanguage()
		{
			grpAutomaticReconnect.Text = My.Language.strGroupboxAutomaticReconnect;
			btnClose.Text = My.Language.strButtonClose;
			lblServerStatus.Text = My.Language.strLabelServerStatus;
			chkReconnectWhenReady.Text = My.Language.strCheckboxReconnectWhenReady;
		}
	}
	
}

File: mRemoteV1/Tools/ReconnectGroup.vb
Imports mRemote3G.My.Resources

Namespace Tools
    Public Class ReconnectGroup
        Private _ServerReady As Boolean

        Public Property ServerReady As Boolean
            Get
                Return _ServerReady
            End Get
            Set
                If value = True Then
                    SetStatusImage(HostStatus_On)
                Else
                    SetStatusImage(HostStatus_Off)
                End If

                _ServerReady = value
            End Set
        End Property

        Private Delegate Sub SetStatusImageCB(Img As Image)

        Private Sub SetStatusImage(Img As Image)
            If pbServerStatus.InvokeRequired Then
                Dim d As New SetStatusImageCB(AddressOf SetStatusImage)
                ParentForm.Invoke(d, New Object() {Img})
            Else
                pbServerStatus.Image = Img
            End If
        End Sub

        Private Sub chkReconnectWhenReady_CheckedChanged(sender As Object, e As EventArgs) _
            Handles chkReconnectWhenReady.CheckedChanged
            _ReconnectWhenReady = chkReconnectWhenReady.Checked
        End Sub

        Private _ReconnectWhenReady As Boolean

        Public Property ReconnectWhenReady As Boolean
            Get
                Return _ReconnectWhenReady
            End Get
            Set
                _ReconnectWhenReady = value
                SetCheckbox(value)
            End Set
        End Property

        Private Delegate Sub SetCheckboxCB(Val As Boolean)

        Private Sub SetCheckbox(Val As Boolean)
            If chkReconnectWhenReady.InvokeRequired Then
                Dim d As New SetCheckboxCB(AddressOf SetCheckbox)
                ParentForm.Invoke(d, New Object() {Val})
            Else
                chkReconnectWhenReady.Checked = Val
            End If
        End Sub

        Public Event CloseClicked()

        Private Sub btnClose_Click(sender As Object, e As EventArgs) Handles btnClose.Click
            RaiseEvent CloseClicked()
        End Sub

        Private Sub tmrAnimation_Tick(sender As Object, e As EventArgs) Handles tmrAnimation.Tick
            Select Case lblAnimation.Text
                Case ""
                    lblAnimation.Text = "»"
                Case "»"
                    lblAnimation.Text = "»»"
                Case "»»"
                    lblAnimation.Text = "»»»"
                Case "»»»"
                    lblAnimation.Text = ""
            End Select
        End Sub

        Private Delegate Sub DisposeReconnectGroupCB()

        Public Sub DisposeReconnectGroup()
            If Me.InvokeRequired Then
                Dim d As New DisposeReconnectGroupCB(AddressOf DisposeReconnectGroup)
                ParentForm.Invoke(d)
            Else
                Me.Dispose()
            End If
        End Sub

        Private Sub ReconnectGroup_Load(sender As Object, e As EventArgs) Handles Me.Load
            ApplyLanguage()
        End Sub

        Private Sub ApplyLanguage()
            grpAutomaticReconnect.Text = Language.Language.strGroupboxAutomaticReconnect
            btnClose.Text = Language.Language.strButtonClose
            lblServerStatus.Text = Language.Language.strLabelServerStatus
            chkReconnectWhenReady.Text = Language.Language.strCheckboxReconnectWhenReady
        End Sub
    End Class
End NameSpace